### PR TITLE
For a while, fishing is working

### PR DESCRIPTION
**Pull request by KennanFatt**
http://minecraft.gamepedia.com/Fishing

### CODE REVIEW

File: src/pocketmine/entity/FishingHook.php [hunk 1]
```diff
@@ -71,19 +72,28 @@ public function onUpdate($currentTick){
 			$this->keepMovement = false;
 			$hasUpdate = true;
 		}
-		if($this->attractTimer === 0 && mt_rand(0, 100) <= 30){ // chance, that a fish bites
-			$this->coughtTimer = mt_rand(5, 10) * 20; // random delay to catch fish
+		if($this->attractTimer === 0 && mt_rand(150, 600) <= 30){ // chance, that a fish bites
+			$this->coughtTimer = mt_rand(1, 1) * 2; // random delay to catch fish
```
Comment (ishitatsuyuki) [inline src/pocketmine/entity/FishingHook.php]: No longer random.

File: src/pocketmine/entity/FishingHook.php [hunk 2]
```diff
@@ -71,19 +72,28 @@ public function onUpdate($currentTick){
 			$this->keepMovement = false;
 			$hasUpdate = true;
 		}
-		if($this->attractTimer === 0 && mt_rand(0, 100) <= 30){ // chance, that a fish bites
-			$this->coughtTimer = mt_rand(5, 10) * 20; // random delay to catch fish
+		if($this->attractTimer === 0 && mt_rand(150, 600) <= 30){ // chance, that a fish bites
```
Comment (ishitatsuyuki) [inline src/pocketmine/entity/FishingHook.php]: What happened to your brain?

File: src/pocketmine/entity/FishingHook.php [hunk 3]
```diff
@@ -71,19 +72,28 @@ public function onUpdate($currentTick){
...
+		if($this->shootingEntity instanceof Player && $this->coughtTimer > 0){
+			$fishes = [ItemItem::RAW_FISH, ItemItem::RAW_SALMON, ItemItem::CLOWN_FISH, ItemItem::PUFFER_FISH, ItemItem::BOW, ItemItem::FISHING_ROD, ItemItem::SADDLE, ItemItem::BOWL, ItemItem::LEATHER, ItemItem::LEATHER_BOOTS, ItemItem::ROTTEN_FLESH, ItemItem::STICK, ItemItem::STRING, ItemItem::BONE, ItemItem::WATER_LILY];
+			$fish = array_rand($fishes, 1);
+			$item = ItemItem::get($fishes[$fish]);
+			$this->getLevel()->getServer()->getPluginManager()->callEvent($ev = new PlayerFishEvent($this->shootingEntity, $item, $this));
+			if(!$ev->isCancelled()){
+				$this->shootingEntity->getInventory()->addItem($item);
+				$this->shootingEntity->addExperience(mt_rand(1, 1));
+				$this->damageRod = true;
+			}
```
Comment (ishitatsuyuki) [inline src/pocketmine/entity/FishingHook.php]: Hey, missing bracket. Test before opening a PR.

**Comment by ishitatsuyuki** [comment]
Closing for horrible code quality. Comment when you have updated your code.

**Comment by KennanFatt** [comment]
i was test it and its work, you can try it, and fix my code placement

**Comment by KennanFatt** [comment]
look at function called reelLine(), that is not working , so i move it

**Comment by Driesboy** [comment]
#dontwork 
#error you forgot }

**Comment by KennanFatt** [comment]
then add }, and try :) okay no problem if you ignore this.. sorry, i just want to tell you how my server can fishing 💨

**Comment by ishitatsuyuki** [comment]
Can you please go to hospital?
Seriously speaking, do you know what's mt_rand and what's the meaning of the parameter?
What you did is too funny.

**Comment by gaoyichuan** [comment]
Your code is so silly... Rejected

**Comment by Muqsit** [comment]
Your code have me cancer.

**Pull request closed without merging**